### PR TITLE
Update annotation functionalities and test

### PR DESCRIPTION
**Pull request by kxyoke**
Addresses #158. `annotation` can now add to general notes.

- [x] Command and parser tests
- [ ] Storage tests

### CODE REVIEW

File: src/test/test.iml [hunk 1]
```diff
@@ -0,0 +1,11 @@
+<?xml version="1.0" encoding="UTF-8"?>
```
Comment (Na-Nazhou) [inline src/test/test.iml]: what is this file for?

Comment (kxyoke) [reply src/test/test.iml]: oh i think it got added when i was configuring test folder?

File: src/main/java/seedu/mark/logic/commands/AnnotationCommand.java [hunk 2]
```diff
@@ -76,3 +95,17 @@ public ParagraphIdentifier getPid() {
...
+ * This is a dummy pid for a non-existent paragraph, null.
+ */
+class DummyParagraphIdentifier extends ParagraphIdentifier {
+    public DummyParagraphIdentifier() {
+        super(Index.fromOneBased(1), ParagraphType.STRAY);
+    }
+
+    @Override
+    public String toString() {
+        return "NULL (i.e. your note has been added to the general ";
```
Comment (dorcastan) [inline src/main/java/seedu/mark/logic/commands/AnnotationCommand.java]: Is this missing a closing bracket? (I can't find where it's used ^^")

File: src/main/java/seedu/mark/logic/commands/EditAnnotationCommand.java [hunk 3]
```diff
@@ -44,7 +44,11 @@
     public static final String MESSAGE_TARGET_NO_PHANTOM = "You cannot move an annotation to a phantom paragraph.";
     public static final String MESSAGE_PHANTOM_CANNOT_HIGHLIGHT = "You cannot change the highlight "
             + "of a phantom paragraph.";
+    public static final String MESSAGE_CANNOT_MOVE_TO_SAME_PARA = "Please specify a different paragraph to move "
+            + "annotation to.";
     public static final String MESSAGE_NOTHING_TO_EDIT = "Paragraph %1$s has no annotations to edit.";
+    public static final String MESSAGE_NO_EDITS_GIVEN = "It seems you forgot to indicate the note or highlight to "
+            + "change! Otherwise there is nothing for me to edit. D:";
```
Comment (dorcastan) [inline src/main/java/seedu/mark/logic/commands/EditAnnotationCommand.java]: This is cute... but maybe we can match the tone of the other messages?

File: src/main/java/seedu/mark/logic/commands/AnnotationCommand.java (hunk 2 above)
Comment (kxyoke) [reply src/main/java/seedu/mark/logic/commands/AnnotationCommand.java]: Oops typo. It's used when the users add to paragraph `p/null`; added the bracket

File: src/main/java/seedu/mark/logic/commands/EditAnnotationCommand.java (hunk 3 above)
Comment (kxyoke) [reply src/main/java/seedu/mark/logic/commands/EditAnnotationCommand.java]: Ok changed!

File: src/main/java/seedu/mark/logic/commands/AnnotationCommand.java [hunk 4]
```diff
@@ -76,3 +95,17 @@ public ParagraphIdentifier getPid() {
...
+ * This is a dummy pid for a non-existent paragraph, null.
+ */
+class DummyParagraphIdentifier extends ParagraphIdentifier {
+    public DummyParagraphIdentifier() {
+        super(Index.fromOneBased(1), ParagraphType.STRAY);
+    }
+
+    @Override
+    public String toString() {
+        return "NULL (i.e. your note has been added to the general)";
```
Comment (dorcastan) [inline src/main/java/seedu/mark/logic/commands/AnnotationCommand.java]:
```suggestion
        return "NULL (i.e. your note has been added to the general notes section)";
```

**Comment by kxyoke** [comment]
> Nice! I like the "Currently showing..." header.
> 
> I feel like the concept of "phantom" and "true" paragraphs is a bit abstract/ might be hard for a new user to understand, especially with other new terms to remember (e.g. remarks vs reminder notes vs general notes etc.). _(For example, to me, 'phantom' paragraphs sounds like 'ghost' paragraphs, which I shouldn't be able to see/ attach things to(?))_ Might there be a more accessible way to describe these two sections of the offline document? Just something to consider!
> 
> On a related note, the paragraph identifiers seem to be different from the description (`S1` for **P**hantom paragraphs but `P1` for **T**rue paragraphs).

Ok! Perhaps I can just rename phantom to be `General notes section`, hence `G1` and original content paragraphs so we can retain the `P1`
Will update in the next pr

**Pull request merged**